### PR TITLE
`ArcGisMapServerCatalogItem` will now use "pre-cached tiles" if no `layers` are specified.

### PR DESCRIPTION
**Pull request by nf-s**
### What this PR does

Cesium's `ArcGisMapServerImageryProvider` has a `usePreCachedTilesIfAvailable` option. This PR sets this is to true if `layers` is undefined and if all `layers` are selected.

#### Add a random MapServer layer that had pre-cached tiles

- http://ci.terria.io/mapserver-emptylayers-precached/
- Add https://services.thelist.tas.gov.au/arcgis/rest/services/Basemaps/Orthophoto/MapServer
- Add first layer to map
- Because this is the same as adding "all layers" - you should now see MapServer use tiled endpoints (in network dev tools)
  - For example https://services.thelist.tas.gov.au/arcgis/rest/services/Basemaps/Orthophoto/MapServer/tile/10/647/930
  - Before it would have looked like - https://services.thelist.tas.gov.au/arcgis/rest/services/Basemaps/Orthophoto/MapServer/export?bbox=17063190.698156465%2C-4852834.0517692715%2C17219733.73208451%2C-4696291.017841227&size=256%2C256&format=png32&transparent=true&f=image&bboxSR=3857&imageSR=3857&layers=show%3A0

#### Add a `esri-mapServer` catalog item with no `layers`

```json
{
      "url": "https://services.thelist.tas.gov.au/arcgis/rest/services/Basemaps/Orthophoto/MapServer/",
      "type": "esri-mapServer",
      "name": "test"
    }
```

- [Before link](http://ci.terria.io/main/#clean&start={%22initSources%22%3A[{%22catalog%22%3A%20[{%22url%22%3A%20%22https%3A%2F%2Fservices.thelist.tas.gov.au%2Farcgis%2Frest%2Fservices%2FBasemaps%2FOrthophoto%2FMapServer%2F%22%2C%22type%22%3A%20%22esri-mapServer%22%2C%22name%22%3A%20%22test%22%2C%22id%22%3A%20%22test%22}]%2C%22workbench%22%3A%20[%22test%22]}]}&share=s-AgA4kYV0z2K3YZmnLcB1Mfy9q3V)
- [After link](http://ci.terria.io/mapserver-emptylayers-precached/#clean&start={%22initSources%22%3A[{%22catalog%22%3A%20[{%22url%22%3A%20%22https%3A%2F%2Fservices.thelist.tas.gov.au%2Farcgis%2Frest%2Fservices%2FBasemaps%2FOrthophoto%2FMapServer%2F%22%2C%22type%22%3A%20%22esri-mapServer%22%2C%22name%22%3A%20%22test%22%2C%22id%22%3A%20%22test%22}]%2C%22workbench%22%3A%20[%22test%22]}]}&share=s-AgA4kYV0z2K3YZmnLcB1Mfy9q3V)

### Checklist

- [x] There are unit tests to verify my changes are correct or unit tests aren't applicable (if so, write quick reason why unit tests don't exist)
- [ ] I've updated relevant documentation in `doc/`.
- [x] I've updated CHANGES.md with what I changed.
- [x] I've provided instructions in the PR description on how to test this PR.

### CODE REVIEW

File: lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts [hunk 1]
```diff
@@ -540,6 +541,8 @@ export default class ArcGisMapServerCatalogItem extends UrlMixin(
         false
       );
 
+      arraysAreEqual;
```
Comment (mwu2018) [inline lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts]: What is it for?

**Review by mwu2018 — CHANGES_REQUESTED** [review]
LGTM except for a nit-pick

File: lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts [hunk 2]
```diff
@@ -552,7 +555,13 @@ export default class ArcGisMapServerCatalogItem extends UrlMixin(
         /** Only used "pre-cached" tiles if we aren't requesting any specific layers
          * If the `layersArray` property is specified, we request individual dynamic layers and ignore the fused map cache.
          */
-        usePreCachedTilesIfAvailable: this.layersArray.length == 0,
+        usePreCachedTilesIfAvailable:
+          this.layersArray.length === 0 ||
+          !this.layers ||
+          arraysAreEqual(
```
Comment (mwu2018) [inline lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts]: If the arrays are only different in order, should still set it to true.

File: lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts (hunk 1 above)
Comment (nf-s) [reply lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts]: Good question! Not sure why I left this in - I'll remove it!

File: lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts (hunk 2 above)
Comment (nf-s) [reply lib/Models/Catalog/Esri/ArcGisMapServerCatalogItem.ts]: Great point, I have changed it to `setsAreEqual`

**Review by mwu2018 — APPROVED** [review]
LGTM

**Pull request merged**